hack/make-rules/test-cmd.sh

```diff
@@ -69,6 +69,7 @@ function run_kube_apiserver() {
     --storage-media-type="${KUBE_TEST_API_STORAGE_TYPE-}" \
     --cert-dir="${TMPDIR:-/tmp/}" \
     --service-cluster-ip-range="10.0.0.0/24" \
+    --client-ca-file=hack/testdata/ca.crt \
```

Comment:
Is there something we could run in the `run_exec_credentials_tests` check to confirm that the client cert auth is enabled on the API server?  I am not sure if this is observable from the outside.  🤔

Reply:
Actually I think you can do something like:

`openssl s_client -connect "127.0.0.1:${API_PORT}" -state`

and check to make sure the output contains:

```
Acceptable client certificate CA names
/CN=127.0.0.1
```

Reply:
Similar approach:

```
$ echo | openssl s_client -connect "127.0.0.1:${API_PORT}" -msg 2>/dev/null | grep CertificateRequest
```

Should output something like:

```
<<< TLS 1.2 Handshake [length 009b], CertificateRequest
```

Reply:
Indeed, the `openssl s_client` trick is what I used to figure out that the client certificate capabilities of the API server was what triggered the exec credential plugin before. Added as a pre-condition.